src/reader/buffered_reader.rs

```diff
@@ -90,11 +90,7 @@ macro_rules! impl_buffered_source {
             }
             *position += read;
 
-            if read == 0 {
-                Ok(None)
-            } else {
-                Ok(Some(&buf[start..]))
-            }
+            Ok((&buf[start..], done))
```

Comment:
Maybe I'm missing something.

This code loops while `!done` and only after the loop has exited is there a potential `Ok` return.  Wouldn't the value of `done` therefore always be `true`?